src/main/java/net/mcreator/ui/workspace/WorkspacePanel.java

```diff
@@ -1181,6 +1212,47 @@ private void addNewFolder() {
 		}
 	}
 
+	private void renameFolder() {
+		renameFolder.setEnabled(true);
+		IElement selected = list.getSelectedValue();
+		String newName = VOptionPane
+				.showInputDialog(mcreator, L10N.t("workspace.elements.folders.rename.message"), L10N.t("workspace.elements.folders.rename.title"), null, new OptionPaneValidatior() {
+					@Override public ValidationResult validate(JComponent component) {
+						String newName = ((JTextField) component).getText();
+						if (!newName.matches("[A-Za-z0-9._ -]+")) {
+							canRename = false;
+							return new Validator.ValidationResult(ValidationResultType.ERROR, L10N.t("workspace.elements.folders.add.error_letters"));
+						} else {
+							canRename = true;
+							return Validator.ValidationResult.PASSED;
+						}
+					}
+				});
+		if (canRename == true) {
+			String finalNewName = newName;
+			list.getSelectedValuesList().forEach(re -> {

```

Comment:
Only one folder of same name can exist so I don't think you need for each as this is sub-optimal compared to normal loop breaking when you find folder with said name.

Reply:
Removed it